hastory-server/test/Data/Hastory/ServerSpec.hs

```diff
@@ -1,83 +1,42 @@
-  , ClientEnv
-  , ClientError(FailureResponse)
-  , Scheme(Http)
-  , mkClientEnv
-  , responseBody
-  , responseStatusCode
-  , runClientM
-  )
-
-data ServerInfo =
-  ServerInfo
-    { siClientEnv :: ClientEnv
-    , siToken :: Token
-    , siPool :: Pool SqlBackend
-    }
+import Servant.Client (ClientError (FailureResponse), responseBody, responseStatusCode, runClientM)
+import Test.Hspec
+import Test.Validity (forAllValid)
 
-testEntry :: Entry
-testEntry = Entry "testText" absDir utcTime "localhost" "testUser"
-  where
-    absDir = [absdir|/|]
-    utcTime = UTCTime (fromGregorian 2020 1 1) (toEnum 0)
+import Data.Hastory.API (Token (..), appendCommand)
+import Data.Hastory.Server.TestUtils (ServerInfo (..), serverSpec)
+import Data.Hastory.Types
+import Data.Time (UTCTime (..), fromGregorian)
 
 spec :: Spec
 spec =
   serverSpec $
   describe "POST /commands/append" $ do
     context "incorrect token" $
       it "is a 403 error with invalid token msg" $ \ServerInfo {..} -> do
-        let req = appendCommand incorrectToken testEntry
+        let testEntry      = Entry "testText" [absdir|/|] utcTime "localhost" "testUser"
+            utcTime        = UTCTime (fromGregorian 2020 1 1) (toEnum 0)
+            req            = appendCommand incorrectToken testEntry
             incorrectToken = Token (unToken siToken <> "badSuffix")
         Left (FailureResponse _ resp) <- runClientM req siClientEnv
         responseBody resp `shouldBe` "Invalid Token provided."
         responseStatusCode resp `shouldBe` status403
     context "with correct token" $
-      it "saves entry to database" $ \ServerInfo {..} -> do
-        _ <- runClientM (appendCommand siToken testEntry) siClientEnv
-        let selectEntries = fmap SQL.entityVal <$> SQL.selectList [] []
-        dbEntries <- SQL.runSqlPool selectEntries siPool
-        length dbEntries `shouldBe` 1
-        head dbEntries `shouldBe` testEntry
-
-serverSpec :: SpecWith ServerInfo -> Spec
-serverSpec = around withTestServer
-
-withTestServer :: (ServerInfo -> IO a) -> IO a
-withTestServer func = do
-  manager <- newManager defaultManagerSettings
-  siToken <- generateToken
-  withSystemTempDir "hastory-server-test" $ \tmpDir -> do
-    filePath <- resolveFile tmpDir "server.db"
-    siPool <- runNoLoggingT $ prepareDb filePath
-    let mkApp = pure $ app opts settings
-        opts = Options 10 Nothing
-        settings = ServerSettings siToken siPool
-    testWithApplication mkApp $ \p ->
-      let siClientEnv = mkClientEnv manager (BaseUrl Http "127.0.0.1" p "")
-       in func (ServerInfo {..})
+      it "saves entry to database" $ \ServerInfo {..} ->
+        forAllValid $ \entry -> do
+            _ <- runClientM (appendCommand siToken entry) siClientEnv
+            let selectEntries = fmap SQL.entityVal <$> SQL.selectList [] []
+            dbEntries <- SQL.runSqlPool selectEntries siPool
+            length dbEntries `shouldBe` 1
+            head dbEntries `shouldBe` entry
```

Comment:
These two lines together can be done using pattern matching instead of using `head`, which is partial.